tests/test_topology.py

```diff
@@ -38,3 +46,118 @@ def test_dual_topology_nonbonded_pairlist():
+    # Implements the full NB potential for the host guest system
+    num_guest_atoms = hgt.guest_topology.get_num_atoms()
+    guest_params, guest_pot = hgt.guest_topology.parameterize_nonbonded(ff_q_params, ff_lj_params, lamb)
+    hg_exclusion_idxs = np.concatenate(
+        [hgt.host_nonbonded.get_exclusion_idxs(), guest_pot.get_exclusion_idxs() + hgt.num_host_atoms]
+    )
+    hg_scale_factors = np.concatenate([hgt.host_nonbonded.get_scale_factors(), guest_pot.get_scale_factors()])
+    hg_nb_params = jnp.concatenate([hgt.host_nonbonded.params, guest_params])
+    return hg_nb_params, potentials.Nonbonded(
+        hgt.num_host_atoms + num_guest_atoms,
+        hg_exclusion_idxs,
+        hg_scale_factors,
+        guest_pot.get_beta(),
+        guest_pot.get_cutoff(),
+    )
+
+
+@pytest.mark.parametrize("precision, rtol, atol", [(np.float64, 1e-8, 1e-8), (np.float32, 1e-4, 5e-4)])
+@pytest.mark.parametrize("ctor", [BaseTopology, DualTopology, DualTopologyMinimization])
+def test_host_guest_nonbonded(ctor, precision, rtol, atol):
+    def compute_ref_grad_u(ff: Forcefield, precision, x0, lamb):
+        # Use the original code to compute the nb grads and potential
+        bt = Topology(ff)
+        hgt = topology.HostGuestTopology(host_bps, bt)
+        params, potentials = parameterize_nonbonded_full(hgt, ff.q_handle.params, ff.lj_handle.params, lamb=lamb)
+        u_impl = potentials.bind(params).bound_impl(precision=precision)
+        return u_impl.execute(x0, solvent_box)
+
+    def compute_split_grad_u(ff: Forcefield, precision, x0, lamb):
+        # Use the updated topology code to compute the nb grads and potential
+        bt = Topology(ff)
+        hgt = topology.HostGuestTopology(host_bps, bt)
+        params, potentials = hgt.parameterize_nonbonded(ff.q_handle.params, ff.lj_handle.params, lamb=lamb)
+        u_impl = potentials.bind(params).bound_impl(precision=precision)
+        return u_impl.execute(x0, solvent_box)
+
+    def compute_vacuum_grad_u(ff: Forcefield, precision, x0, lamb):
+        # Compute the vacuum nb grads and potential
+        bt = Topology(ff)
+        params, potentials = bt.parameterize_nonbonded(ff.q_handle.params, ff.lj_handle.params, lamb=lamb)
+        u_impl = potentials.bind(params).bound_impl(precision=precision)
+        return u_impl.execute(x0, solvent_box)
+
+    ff_ref, ff_split, ff_scaled, ff_inter_scaled = load_split_forcefields()
+
+    box_width = 4.0
+    solvent_sys, solvent_conf, solvent_box, solvent_top = build_water_system(box_width, ff_ref.water_ff)
+    solvent_box += np.diag([0.1, 0.1, 0.1])
+    host_bps, host_masses = openmm_deserializer.deserialize_system(solvent_sys, cutoff=1.2)
+
+    with resources.path("timemachine.testsystems.data", "ligands_40.sdf") as path_to_ligand:
+        all_mols = read_sdf(path_to_ligand)
+
+    if ctor == BaseTopology:
+        mol = all_mols[1]
```

Comment:
nit: would it be preferable to lookup by mol name here? (maybe not an issue if we're confident ordering is deterministic)